packages/flutter_tools/lib/src/migrate/migrate_compute.dart

```diff
@@ -0,0 +1,620 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/terminal.dart';
+import '../flutter_project_metadata.dart';
+import '../project.dart';
+import '../version.dart';
+import 'custom_merge.dart';
+import 'migrate_result.dart';
+import 'migrate_utils.dart';
+
+// This defines files and directories that should be skipped regardless
+// of gitignore and config settings
+const List<String> _skippedFiles = <String>[
+  'lib/main.dart', // Almost always user owned.
+  'ios/Runner.xcodeproj/project.pbxproj', // Xcode managed configs that may not merge cleanly.
+  'README.md', // changes to this shouldn't be overwritten since is is user owned.
+];
+
+const List<String> _skippedDirectories = <String>[
+  '.dart_tool', // ignore the .dart_tool generated dir
+  '.git', // ignore the git metadata
+  'lib', // Files here are always user owned and we don't want to overwrite their apps.
+  'test', // Files here are typically user owned and flutter-side changes are not relevant.
+  'assets', // Common directory for user assets.
+];
+
+bool _skipped(String localPath, {Set<String?>? blacklistPrefixes}) {
```

Comment:
```suggestion
bool _skipped(String localPath, {Set<String?>? denylistPrefixes}) {
```

Reply:
I made it non-nullable. Not sure if I agree with blacklist->denylist though, I think blacklist is a bit more consistent with common terminology.